workspaces/optic-engine/src/queries/shape.rs

```diff
@@ -96,23 +96,16 @@ impl<'a> ShapeQueries<'a> {
               .shape_parameter_id;
             let item_shape_id =
               self.resolve_parameter_to_shape(&shape_id, &String::from(optional_parameter_id));
-            let trail = shape_trail
+            let item_trail = shape_trail
               .with_component(ShapeTrailPathComponent::OptionalTrail {
                 shape_id: shape_id.clone(),
               })
               .with_component(ShapeTrailPathComponent::OptionalItemTrail {
                 shape_id: shape_id.clone(),
                 inner_shape_id: item_shape_id.clone(),
               });
-            let mut output = vec![ChoiceOutput {
-              parent_trail: shape_trail.clone(),
-              additional_components: vec![ShapeTrailPathComponent::OptionalTrail {
-                shape_id: shape_id.clone(),
-              }],
-              shape_id: shape_id.clone(),
-              core_shape_kind: core_shape_node.descriptor.kind.clone(),
-            }];
-            output.append(&mut self.list_trail_choices(&trail));
```

Comment:
The choice of only the `OptionalTrail`, without any description of what the underling optional shape is, is not actually a useful one in the diffing, and was the main source of this bug. I've yet to find anything else beyond the shape choice mapping that relied on this, but I'll dig in deeper through testing to make sure.

Reply:
the contract of list_trail_choices is to list the choices at a particular trail. Thus the optional trail has meaning, communicating that it's allowed for a value to not be present here. When you compute coverage, for example, you would need to know that to track how often the value is omitted.

Reply:
> Thus the optional trail has meaning, communicating that it's allowed for a value to not be present her

I guess what I'm trying to assert here is that "optional" is an incomplete choice. You'll want to know _what_ is the thing that is optional for the choice to be complete. There is a 1:1 mapping here, we don't have the case of an optional, without knowing what the subject of the optionality is. Therefor, I think it's right to only return the "full" choice: optional, plus it's subject. That should still be enough to compute coverage from, right? Count `OptionalTrailItem` instead of `OptionalTrail` as the two should always come together?

Reply:
Optional means there are 2 possibilities: providing nothing (undefined) or providing one of the choices afforded by the optional item. You don't need to know what the optional item is to know you can provide nothing. Similarly, if we instead had no Optional but had Undefined, we would use OneOf(Undefined, ...) and the Undefined would be one of the choices. I'm not understanding what you're trying to say about it being incomplete and something that list_trail_choices shouldn't emit

Reply:
Coverage counting when it is undefined should increment that Optional shapeid's trail and coverage counting any of the items should count their trails. Each possibly outcome should have its own trail, right?

Reply:
We got aligned on this in a sync call, with as a conclusion that `list_trail_choices` _should_ contain both choices as they strictly _are_ choices. Instead, we'll deal with some choices not being meaningful in certain contexts downstream.